react/week3/TodoDescription.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+function TodoDescription({ addTodo }) {
+  const [state, setState] = useState("");
+  const [date, setDate] = useState(new Date());
+
+  const onFormSubmit = event => {
+    event.preventDefault();
+    if (!state || !date) {
+      return alert("Please enter details");
+    }
+    addTodo(state, date.toISOString().substr(0, 10));
+    setState("");
+    setDate("");
+  };
+  const handleChange = date => {
+    setDate(date);
+  };
+
+  return (
+    <form onSubmit={onFormSubmit}>
+      <label>
+        Todo Description :
+        <input
+          type="text"
+          name="description"
+          value={state}
+          onChange={event => {
+            setState(event.target.value);
+          }}
+        ></input>
+      </label>
+      <label id="datePicker">
+        DeadLine:
+        <DatePicker selected={date} onChange={handleChange} />

```

Comment:
No need for a separate function if we only update the state
```suggestion
        <DatePicker selected={date} onChange={date => setDate(date)} />

```